airdrop/application/services/airdrop_services.py

```diff
@@ -48,20 +51,28 @@ def get_txn_receipt(self, txn_hash):
             print(f"Exception on get_txn_receipt {e}")
             raise e
 
-    def airdrop_listen_to_events(self, event):
+    def airdrop_event_consumer(self, event):
+        response = {}
+        status = HTTPStatus.BAD_REQUEST
+
         event_data = event['data']
         event_name = event_data['event']
 
         if event_name == AirdropEvents.AIRDROP_CLAIM.value:
-            return self.update_airdrop_window_claim_status(event_data)
+            self.update_airdrop_window_claim_status(event_data)
+            status = HTTPStatus.OK
+        else:
+            response = "Unsupported event"
+
+        return status, response
 
     def update_airdrop_window_claim_status(self, event):
         try:
-            event_payload = event['json_str']
+            event_payload = ast.literal_eval(event["json_str"])
             user_address = event_payload['claimer']
-            amount = int(event_payload['amount'])
-            airdrop_id = str(event_payload['airDropId'])
-            airdrop_window_id = str(event_payload['airDropWindowId'])
+            amount = event_payload['amount']
+            airdrop_id = event_payload['airDropId']
+            airdrop_window_id = event_payload['airDropWindowId']
             txn_hash = event['transactionHash']
             txn_status = AirdropClaimStatus.SUCCESS.value
             AirdropRepository().create_or_update_txn(
```

Comment:
@rajeevbbqq we should handle the scenario of dropped transactions. If we find that the transaction in the event is not in the table we need to 
- Check if there exists a transaction in pending state that has the same address, airdrop ID, airdrop window ID.
- If we find such an entry validate that the transaction hash of that entry has been dropped and update the row

Reply:
We can create a new issue for that and fix it in a different PR

Reply:
@raamb we are handling that scenario please find the screenshots below from DB.
We have a corn job which watches the transaction is dropped or not and update in DB

![Screenshot from 2021-12-28 14-49-16](https://user-images.githubusercontent.com/16451615/147550769-87f17284-8cf1-4fec-922a-71454f32e5cc.png)

![Screenshot from 2021-12-28 14-46-20](https://user-images.githubusercontent.com/16451615/147550764-49cb027c-3cb2-4749-a899-b84ec6f12152.png)
o